src/collective/ifttt/utils.py

```diff
@@ -0,0 +1,158 @@
+# -*- coding: utf-8 -*-
+
+from Acquisition import aq_parent
+from collective.ifttt import _
+from plone.app.contentrules import api as rules_api
+from plone.contentrules.engine.interfaces import IRuleStorage
+from plone.contentrules.rule.interfaces import IRuleAction
+from plone.contentrules.rule.interfaces import IRuleCondition
+from Products.CMFCore.interfaces._events import IActionSucceededEvent
+from zope.component import getMultiAdapter
+from zope.component import getUtility
+
+
+class Rules(object):
+    """Define content rule operations within collective.ifttt environment"""
+
+    def __init__(self, context, request):
+        self.context = context
+        self.request = request
+        '''
+        data should have keys
+        ifttt_event_name, content_types, workflow_transitions  and payload
+        '''
+
+    def add_rule(self, data):
+        '''
+        Create new rule
+        require data related to field values to create new rule
+        '''
+
+        # REDFINE ME
+        rule_name = _(
+            u'${title}_Trigger_${ifttt_event_name}',
+            mapping=dict(
+                ifttt_event_name=data.get('ifttt_event_name'),
+                title=self.context.Title().decode('utf-8', 'ignore'),
+            )
+        )
+
+        rule_description = _(
+            u'This rule is created to trigger ${ifttt_event_name} '
+            u'on ${title} folder',
+            mapping=dict(
+                ifttt_event_name=data.get('ifttt_event_name'),
+                title=self.context.Title().decode('utf-8', 'ignore'),
+            )
+        )
+
+        # HACK
+        adding = self.context.restrictedTraverse('/Plone/+rule')
```

Comment:
@Shriyanshagro You cannot assume that the portal root name is `Plone`.

You can get the portal object 

```python
portal = api.portal.get()
adding = portal.restrictedTraverse('+rule')
```

But possibly that is also available through adapter lookup (faster)

```python
adding = getMultiAdapter((portal, self.request), name='+rule')
```

Reply:
Later you use multiadapter lookups, so it is also the right solution here.

This is probably also the reason why acceptance tests fail.